File: wiki_java/src/main/java/com/github/klee0kai/wiki/wrapping/TechFactoryModule.java
package com.github.klee0kai.wiki.wrapping;

import com.github.klee0kai.stone.annotations.module.Module;
import com.github.klee0kai.stone.annotations.module.Provide;
import com.github.klee0kai.test.tech.components.Battery;
import com.github.klee0kai.test.tech.components.Ram;


@Module
public interface TechFactoryModule {

    @Provide(cache = Provide.CacheType.Factory)
    Battery battery();

    @Provide(cache = Provide.CacheType.Factory)
    Ram ram();

}

File: wiki_kotlin/src/main/kotlin/com/github/klee0kai/wiki/wrapping/TechFactoryModule.kt
package com.github.klee0kai.wiki.wrapping

import com.github.klee0kai.stone.annotations.module.Module
import com.github.klee0kai.stone.annotations.module.Provide
import com.github.klee0kai.test.tech.components.Battery
import com.github.klee0kai.test.tech.components.Ram

@Module
interface TechFactoryModule {

    @Provide(cache = Provide.CacheType.Factory)
    fun battery(): Battery

    @Provide(cache = Provide.CacheType.Factory)
    fun ram(): Ram

}
